lib/dune

```diff
@@ -0,0 +1,18 @@
+(library
+  (name oS)
+  (public_name mirage-solo5.internals)
+  (modules lifecycle main mM oS solo5 time)
+  (c_names alloc_pages_stubs atomic_stubs barrier_stubs checksum_stubs clock_stubs cstruct_stubs main mm_stubs solo5_block_stubs solo5_console_stubs solo5_net_stubs)
+  (c_flags (:include cflags) -O2 -std=c99 -Wall -Werror)
```

Comment:
I realise that the -Werror is existing behaviour, but it makes it really painful when doing production builds with a newer compiler that introduces some random warning.

Reply:
I share your concerns, but IMHO such a change (removing `-Werror`) deserves its own PR rather than being included (hidden) in a build-system-changing PR. Out of curiousity, is it possible in dune to specify specific `development` vs `release` c_flags (so there could be `-Werror` in development mode)?